### PR TITLE
Add current page parameter to the route in the listing and search block pagination

### PR DESCRIPTION
**Pull request by bipoza**
Added the current page parameter to the route in the listing and search block pagination.

Fixes: #3868 

## Demo with listing block
https://user-images.githubusercontent.com/26112509/208886147-6c51f379-3be2-4e83-9743-d3d9abb823e4.mov

## Demo with search block
https://user-images.githubusercontent.com/26112509/208886239-67a762a6-8170-49ac-8ae6-4858c0afac11.mov

## Demo combining multiple lists
https://user-images.githubusercontent.com/26112509/208963510-32668c24-76f0-4d72-be96-16766aa2e3c0.mov

## QueryString compatibility:
To be able to combine the searchBlock with the pagination parameters, we had to change the parameters of the searchBlock from hash to search (# -> ?).

We are contemplating the possibility of maintaining both types of routes so as not to generate a breaking change.

In general the compatibility is good except for a small error in the design of the SearchInput. With the old hash url, the clear icon does not render well, although the functionality is correct.

https://user-images.githubusercontent.com/26112509/208971981-2bc180ac-9c3a-4228-988f-cf239fe8081f.mov

### CODE REVIEW

**Comment by ionlizarazu** [comment]
The bug you mention exists in the current (16.4.0) Volto version

**Review by erral — APPROVED** [review]

**Comment by sneridagh** [comment]
@bipoza @erral wow! thanks for working in this. I know is specially difficult to accomplish. How are we maintaining the sync with the state-URL? I know this is a major pain point. Could we have unit tests for this feature? How about acceptance tests?

What we are trying to accomplish here is quite complex and I'd like to discuss this in the next Volto Team meeting with the team, and specially @tiberiuichim who tried to implement this in the past. 

I'd like to get this well, specially agree on the format and serialization of the URL query parameters, because if we have to change them in the future for whatever reason, it will force a breaking change (people having saved URLs as bookmarks, etc)

**Comment by sneridagh** [comment]
@bipoza the towncrier file extension is wrong, could you please amend it? I think it's ready to merge since long time ago. I have doubts if it's a bugfix or a feature...

**Comment by erral** [comment]
We have seen that we have a bug in the search block with this PR.

We are preparing a fix and some tests to improve the PR.

File: src/helpers/Utils/usePagination.js [hunk 1]
```diff
@@ -24,26 +27,29 @@ const useCreatePageQueryStringKey = (id) => {
  * A pagination helper that tracks the query and resets pagination in case the
  * query changes.
  */
-export const usePagination = (id = null, defaultPage = 1) => {
+export const usePagination = (blockId = null, defaultPage = '1') => {
```
Comment (tiberiuichim) [inline src/helpers/Utils/usePagination.js:30]: I'd think we should stick to a generic `id` rather then `blockId` and a number for the defaultPage.

Comment (ionlizarazu) [reply src/helpers/Utils/usePagination.js:30]: You mean the "blockId" naming or something else?

**Comment by ionlizarazu** [comment]
We have written some tests for the usePagination.js file

**Comment by erral** [comment]
@sneridagh we think this is ready

**Pull request merged**